### DISCUSSION TITLE
Reference project metadata from `quarto.yml`

### DISCUSSION THREAD

**Discussion by pat-s** [quarto-dev/quarto-cli#1072] [Q&A]
https://quarto.org/docs/authoring/shortcodes.html explains how to reference metadata from the active document. 

Is there a way to reference metadata from `_quarto.yml`? E.g. the `title` or `author` field would be very helpful 🙂

**Comment by cderv** [comment]
@pat-s I believe this should already be possible. From https://quarto.org/docs/authoring/variables.html#meta
> The meta shortcode allows you to insert content from Pandoc metadata (e.g. YAML at the top of the document and/or in _quarto.yml).

Do you have a specific meta field that is not working ? 

I tried used in a website project file `{{< meta website.title >}}`  and it correctly got me the website title. 

thanks

**Reply by pat-s** [reply]
Thanks!
This helped for `book.title` but it does not seem to support lists as in `book.author` (with multiple authors) :(

```
WARNING: Unsupported type 'List' for key book.author in a meta shortcode.
```

Do I need to defined a custom shortcode for this?

Specifically, I am trying to get the following working in a dynamic way: 

```
@misc{
  title = {{< meta book.title >}},
  author = {{< meta book.author >}}
  [...]
}
```

One problem here is that if I want to enclose the shortcode within `{}`, I can't find a way to get rid of the whitespace. 
To make it more clear, the following works

```
{ {{< meta book.title >}} }
```

while 

```
{{{< meta book.title >}}}
```

does not. I haven't found a way to properly escape things here or get rid of the whitespace introduced in the above example.

**Comment by jjallaire** [comment]
Yes, metadata is merged from `_project.yml` and should be fully available to the `meta` shortcode

**Comment by jjallaire** [comment]
Unfortunately there isn't a way to do this right now (the `meta` shortcode just deals with plain strings). You would probably need to write a custom pandoc filter that traverses the actual `Meta` passed to Pandoc.
